### PR TITLE
Restore 3.9 compatibility by replacing | with Union[]

### PR DESCRIPTION
**Pull request by lstein**
This PR replaces all places where `footype | bartype` was being used to specify alternative types with `Union[footype, bartype]`. This restores the ability of InvokeAI to run on Python 3.9.

### CODE REVIEW

**Comment by blessedcoolant** [comment]
Is there anything that is keeping us on 3.9? 3.10 has a lot nice new features.

File: invokeai/app/cli/commands.py [hunk 1]
```diff
@@ -47,7 +47,7 @@ def add_parsers(
     commands: list[type],
     command_field: str = "type",
     exclude_fields: list[str] = ["id", "type"],
-    add_arguments: Callable[[argparse.ArgumentParser], None]|None = None
+    add_arguments: Union[Callable[[argparse.ArgumentParser], None],None] = None
```
Comment (StAlKeR7779) [inline invokeai/app/cli/commands.py:50]:
```suggestion
    add_arguments: Optional[Callable[[argparse.ArgumentParser], None]] = None
```
Note: need import Optional from typing

File: invokeai/app/cli/commands.py [hunk 2]
```diff
@@ -72,7 +72,7 @@ def add_parsers(
 def add_graph_parsers(
     subparsers,
     graphs: list[LibraryGraph],
-    add_arguments: Callable[[argparse.ArgumentParser], None]|None = None
+    add_arguments: Union[Callable[[argparse.ArgumentParser], None], None] = None
```
Comment (StAlKeR7779) [inline invokeai/app/cli/commands.py:75]:
```suggestion
    add_arguments: Optional[Callable[[argparse.ArgumentParser], None]] = None
```
Note: need import Optional from typing

File: invokeai/backend/model_management/model_probe.py [hunk 3]
```diff
@@ -64,7 +64,7 @@ def heuristic_probe(cls,
     @classmethod
     def probe(cls,
               model_path: Path,
-              model: Union[Dict, ModelMixin] = None,
+              model: Optional[Union[Dict, ModelMixin]],
```
Comment (StAlKeR7779) [inline invokeai/backend/model_management/model_probe.py:67]: You sure that there no need in default argument?
On 58 line:
```python
return cls.probe(model_path=model,prediction_type_helper=prediction_type_helper)
```
it called without model argument and with current code should raise and exception.

**Review by StAlKeR7779 — CHANGES_REQUESTED** [review]

File: invokeai/backend/model_management/model_probe.py (hunk 3 above)
Comment (lstein) [reply invokeai/backend/model_management/model_probe.py:67]: Actually I caught that earlier today and fixed it.

**Comment by hipsterusername** [comment]
@lstein @StAlKeR7779 is this still needed?

**Pull request closed without merging**